MLKHelperGUI2/Program.cs

```diff
@@ -25,7 +18,27 @@ static void Main(string[] args)
             {
                 Application.EnableVisualStyles();
                 Application.SetCompatibleTextRenderingDefault(false);
-                if (args.Length == 0) { MessageBox.Show("No .MLK, no Muffins!"); }
+
+                var cHelper = new HelperClass();
+                if (!cHelper.InstallLocationInRegistryKeyExists())
+                {
+                    if (!cHelper.IsAdministrator()) //Needs to run as admin to edit the registry
+                    {
+                        cHelper.StartProcessAsAdmin(cHelper.SelfPath, args.Length > 0 ? args[0] : "");
+                        Environment.Exit(0);
+                    }
+
+                    if (MessageBox.Show(".MLK files aren't associated with this program. Set association now?", "MLK song installer", MessageBoxButtons.YesNo, MessageBoxIcon.Question) == DialogResult.Yes)
```

Comment:
this might get really annoying for portable installs. not sure if we want this.
@yaymuffins @eldan-dex please chat about this in irc